src/fheroes2/dialog/dialog_language_selection.cpp

```diff
@@ -200,23 +255,25 @@ namespace
                 fheroes2::showMessage( header, body, 0 );
             }
 
-            if ( items.isSelected() ) {
-                const fheroes2::SupportedLanguage newChosenLanguage = items.GetCurrent();
+            if ( !listBox.IsNeedRedraw() && !needRedraw ) {
```

Comment:
We can simplify this check: `needRedraw` can be `true` only after `_currentId` is changed during the `QueueEventProcessing();`. And if it is changed then `listBox.IsNeedRedraw()` will be true.
So there is no need to check `needRedraw` here.
```suggestion
            if ( !listBox.IsNeedRedraw() ) {
```